ddprof-lib/src/main/cpp/wallClock.cpp

```diff
@@ -174,6 +175,14 @@ void WallClockJVMTI::timerLoop() {
         bool do_filter = Profiler::instance()->threadFilter()->enabled();
         int self = OS::threadId();
 
+        // If filtering is enabled, collect the filtered TIDs first
+        std::vector<int> filtered_tids;
+        if (do_filter) {
+            Profiler::instance()->threadFilter()->collect(filtered_tids);
+            // Sort the TIDs for efficient lookup
+            std::sort(filtered_tids.begin(), filtered_tids.end());
```

Comment:
todo: bench this part if useful. the collection is not idea. We could really randomize during the collection.